src/item.cpp

```diff
@@ -2298,6 +2245,69 @@ std::string item::display_name(unsigned int quantity) const
     return string_format("%s%s%s", name.c_str(), side.c_str(), qty.c_str());
 }
 
+std::string item::get_item_tags(bool parens) const
+{
+    std::string rtn;
+    std::stringstream ret;
+    ret.str("");
```

Comment:
Rather than an option I'd prefer for the new menu to be good enough to be
the only menu, supporting multiple menu layouts adds significant
maintenance overhead.
That said, please take some time to outline how the new menu is better.
It's difficult to review your code for correctness if we don't know your
specific goals.